beacon_node/http_api/src/lib.rs

```diff
@@ -1715,6 +1715,12 @@ pub fn serve<T: BeaconChainTypes>(
                     let indices = indices_res?;
                     let blob_sidecar_list_filtered =
                         block_id.blob_sidecar_list_filtered(indices, &chain)?;
+                    let (block, execution_optimistic, finalized) =
+                        block_id.blinded_block(&chain)?;
```

Comment:
This is a bit inefficient because we end up computing the block root twice, once when loading the blobs and then again when loading the block. I think this is probably an acceptable performance tradeoff for now, but maybe we could add a comment here noting that this could be optimised further if desired

Reply:
updated to only compute the block root once [a7c60a4](https://github.com/sigp/lighthouse/pull/5987/commits/a7c60a448a60aff6d502824af4f64f28834d6f1f)

Reply:
Would it be more efficient if we get the root first with 

```
let root = self.root(chain)?.0;
```

and then use this root for both `chain.get_blinded_block` and `chain.get_blobs`, so we don't have to do `block.message.tree_hash_root()`?

Reply:
added in 969e4c7fa76b4bb90afa714e90530638bd492d08